### PR TITLE
fix for checking FMA_NATIVE

### PR DESCRIPTION
**Pull request by KristofferC**
I guess this needs to be run inside a compiled function to not get run by the interpreter or something along those lines?

```jl
julia> Base.Math.FMA_NATIVE
false

julia> muladd(nextfloat(1.0),nextfloat(1.0),-nextfloat(1.0,2))
4.930380657631324e-32

julia> (() -> muladd(nextfloat(1.0),nextfloat(1.0),-nextfloat(1.0,2)))()
0.0

julia> @code_native debuginfo=:none muladd(1.0, 2.0, 3.0)
        .section        __TEXT,__text,regular,pure_instructions
        vfmadd213sd     %xmm2, %xmm1, %xmm0  # <--------- yes, I have it
        retq
        nopw    %cs:(%rax,%rax)
```

Might improve performance for `hypot` and `log` in some cases.

### CODE REVIEW

File: base/special/log.jl [hunk 1]
```diff
@@ -141,7 +141,8 @@ const t_log_Float32 = (0.0,0.007782140442054949,0.015504186535965254,0.023167059
 
 # determine if hardware FMA is available
 # should probably check with LLVM, see #9855.
-const FMA_NATIVE = muladd(nextfloat(1.0),nextfloat(1.0),-nextfloat(1.0,2)) != 0
+# needs to be run inside a compiled function
+const FMA_NATIVE = (() -> muladd(nextfloat(1.0),nextfloat(1.0),-nextfloat(1.0,2)))()
```
Comment (mbauman) [inline base/special/log.jl]:
```suggestion
const FMA_NATIVE = (() -> muladd(nextfloat(1.0),nextfloat(1.0),-nextfloat(1.0,2)))() != 0
```

Comment (KristofferC) [reply base/special/log.jl]: 🤣

**Comment by mbauman** [comment]
This behaves strangely at the REPL, too.

```julia
# Machine without fast FMA
julia> muladd(nextfloat(1.0),nextfloat(1.0),-nextfloat(1.0,2))
0.0

julia> (() -> muladd(nextfloat(1.0),nextfloat(1.0),-nextfloat(1.0,2)))()
0.0

# Machine with fast FMA
julia> muladd(nextfloat(1.0),nextfloat(1.0),-nextfloat(1.0,2))
4.930380657631324e-32

julia> (() -> muladd(nextfloat(1.0),nextfloat(1.0),-nextfloat(1.0,2)))()
0.0
```

**Comment by KristofferC** [comment]
I think I confused myself here since I somehow thought the check was for `== 0`... So yeah, this doesn't really make sense since it still gives the wrong answer.

**Comment by yuyichao** [comment]
This test is wrong. Llvm has all the freedom it want to return either values for muladd no matter if fma is available. Fwiw, arm / aarch 64 even have fast muladd instruction with intermediate rounding.

**Comment by KristofferC** [comment]
Thanks @yuyichao, closing this and opening an issue instead.

**Pull request closed without merging**